nixos/modules/services/misc/sonarr.nix

```
@@ -44,6 +83,15 @@ in
   };
 
   config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.settings ? ApiKey -> false;
+        message = ''
+          The `services.sonarr.settings` attribute set must not contain `ApiKey`, as it is a secret and cannot be securely stored in the Nix store.
```

This should point to `apiKeyFile` in the message, how about:

```suggestion
          The `services.sonarr.settings` attribute set must not contain `ApiKey`, you should instead use `services.sonarr.apiKeyFile` to avoid storing secrets in the Nix store.
```

---

done